File: KMPSampleIOSApp/KMPSampleIOSApp/utils/PriorityQueueUtil.swift
import Foundation
import KMPSampleSwiftPackage

enum PriorityQueueUtil {
    static func pickMinimum(from array: [Int]) -> Int? {
        let pq = PriorityQueue<NSNumber>(
            comparator: { a, b in a.int32Value - b.int32Value }
        )
        array.forEach { pq.push(NSNumber(integerLiteral: $0)) }
        return pq.pop()?.intValue
    }
}

File: KMPSampleAndroidApp/app/src/main/java/dev/kaseken/kmpsampleandroidapp/utils/PriorityQueueUtil.kt
package dev.kaseken.kmpsampleandroidapp.utils

import dev.kaseken.kmpsamplelibrary.PriorityQueue

object PriorityQueueUtil {
  fun pickMinimum(array: List<Int>): Int? {
    val pq = PriorityQueue<Int> { a, b -> a - b }
    array.forEach { pq.push(it) }
    return pq.pop()
  }
}
